src/UI/Button.elm

```diff
@@ -1,30 +1,73 @@
 module UI.Button exposing
-    ( Button
-    , ButtonBody
-    , ButtonStyle
-    , ButtonWidth
-    , clear
-    , cmd
-    , danger
-    , disabled
-    , fromIcon
-    , fromLabel
-    , full
-    , hyperlink
-    , light
-    , map
-    , primary
-    , redirect
-    , renderElement
-    , shrink
-    , success
-    , toggle
-    , withDisabledIf
+    ( Button, toggle, success, disabled, cmd, redirect
+    , ButtonBody, fromLabel, fromIcon
+    , ButtonStyle, hyperlink, primary, danger, light, clear
+    , ButtonWidth, withWidth, full, shrink
     , withSize
-    , withSuccessIf
-    , withWidth
+    , withDisabledIf, withSuccessIf
+    , renderElement
+    , map
     )
 
+{-| The button is the unified aggregation of all the standardized clickable elements.
```

Comment:
Maybe I don't need to expose that we treat the link and embossed button are different way. we could only say:

"The same component can render as a hyperlink, primary, danger, light and clear"